### ISSUE TITLE
Identify incorrect MessageVisibilityTimeout using EventSourceMapping

### ISSUE THREAD

**Issue by abhinav-enboarder** [aws-cloudformation/cfn-lint#3987]
### CloudFormation Lint Version

1.25.0

### What operating system are you using?

MacOS

### Describe the bug

When creating an SQS queue, Lambda function and an EventSourceMapping that connects the two, we do not get feedback about the message visibility timeout value.

### Expected behavior

If the message visibility timeout is less than lambda function's timeout duration, EventSourceMapping creation should error.

### Reproduction template

```yaml
MyFifoQueue:
    Type: AWS::SQS::Queue
    Properties:
      QueueName: MyQueue.fifo
      FifoQueue: true
      ContentBasedDeduplication: true
      VisibilityTimeout: 120 # should be 300 or more
      MessageRetentionPeriod: 7200

SQSBatch:
    Type: AWS::Lambda::EventSourceMapping
    Properties:
      BatchSize: 10
      Enabled: true
      EventSourceArn: !GetAtt MyFifoQueue.Arn
      FunctionName: !Ref Lambda

Lambda:
    Type: AWS::Lambda::Function
    Properties:
      Code:
        S3Bucket: !Sub "${S3Bucket}${AWS::Region}"
        S3Key: !Sub "${Project}/${App}/${Branch}/${Timestamp}/function.zip"
      Role: !ImportValue all-roles:MyRoleArn
      FunctionName: MyFunction
      Timeout: 300
      Handler: index.handler
      Runtime: !Ref LambdaRunTime
      MemorySize: 256

```
